x/curating/abci.go

```diff
@@ -93,3 +98,22 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) []abci.ValidatorUpdate {
 	}
 	return []abci.ValidatorUpdate{}
 }
+
+// sends matching reward in proportion to upvote
+func sendMatchingReward(ctx sdk.Context, k keeper.Keeper, upvoteAmount sdk.Int,
+	matchPoolPerVote sdk.Dec, rewardAccount sdk.AccAddress) error {
+
+	voteNum, err := upvoteAmount.QuoRaw(1_000_000).ToDec().ApproxSqrt()
```

Comment:
I think this assumes amount has a relation of `1:1`, but don't think this considers the quadratic votes

Reply:
Let's walk through a case: If upvote is 9 ATOM, voteNum should be 3.
```
// because we store the coin in micros, the upvote amount is...
upvoteAmount = 9_000_000
// so we need to divide by 1,000,000 to normalize
temp = 9
// now we take the square root to come up with the original voteNum
voteNum = 3
```
How did you get 9?

Reply:
doh yeah ignore me is saturday night lol sounds good, I just noticed that we didn't store voteNum in the upvote struct  and got confused. 

Not sure if it would be better to store the voteNum inside each upvote as well, since the default VoteAmount is a Param that could potentially change (gov change required tho)

Reply:
Yeah good point. I was trying to be too efficient and not store it, but that was premature optimization. We should store it: https://github.com/public-awesome/stakebird/issues/245.

Reply:
Also 🍺:)